core/banking/recurring_transfers.go

```diff
@@ -183,27 +243,44 @@ func (e *Engine) distributeRecurringTransfers(ctx context.Context, newEpoch uint
 		var r []*types.LedgerMovement
 		if v.DispatchStrategy == nil {
 			resps, err = e.processTransfer(
-				ctx, v.From, v.To, v.Asset, "", "", v.FromAccountType, v.ToAccountType, amount, v.Reference, nil, // last is eventual oneoff, which this is not
+				ctx, v.From, v.To, v.Asset, "", v.FromAccountType, v.ToAccountType, amount, v.Reference, nil, // last is eventual oneoff, which this is not
 			)
 		} else {
 			// check if the amount + fees can be covered by the party issuing the transfer
 			if _, err = e.ensureFeeForTransferFunds(amount, v.From, v.Asset, v.FromAccountType); err == nil {
-				marketScores := e.getMarketScores(v.DispatchStrategy, v.Asset, v.From)
-				// first we make sure that there's sufficient funds to cover the transfer
-				for _, fms := range marketScores {
-					amt, _ := num.UintFromDecimal(amount.ToDecimal().Mul(fms.Score))
-					if amt.IsZero() {
-						continue
+				// NB: if the metric is market value we're going to transfer the bonus if any directly
+				// to the market account of the asset/reward type - this is similar to previous behaviour and
+				// different to how all other metric based rewards behave. The reason is that we need the context of the funder
+				// and this context is lost when the transfer has already gone through
+				if v.DispatchStrategy.Metric == vegapb.DispatchMetric_DISPATCH_METRIC_MARKET_VALUE {
+					marketProposersScore := e.marketActivityTracker.GetMarketsWithEligibleProposer(v.DispatchStrategy.AssetForMetric, v.DispatchStrategy.Markets, v.Asset, v.From)
+					for _, fms := range marketProposersScore {
+						amt, _ := num.UintFromDecimal(amount.ToDecimal().Mul(fms.Score))
+						if amt.IsZero() {
+							continue
+						}
+						r, err = e.processTransfer(
+							ctx, v.From, v.To, v.Asset, fms.Market, v.FromAccountType, v.ToAccountType, amt, v.Reference, nil, // last is eventual oneoff, which this is not
+						)
+						if err != nil {
+							e.log.Error("failed to process transfer", logging.Error(err))
```

Comment:
What about adding a bit more context on the error log?
- from
- to
- asset
- amount


So if there is a problem we can debug a bit more easily. The underlying error doesn't seem to embed enough information.